### PR TITLE
Added StopPlay for refbox Stop gamestate

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Renamed the existing StopPlay to HaltPlay, and created a new StopPlay to be used during the refbox "Stop" gamestate. This play is supposed to limit robot speed to 1.5m/s, and keep robots at least 0.5m away from the ball. Currently, this play lines up robots in front of the defense area.

Future action items are to see if we really have a way to limit robot speed (currently we have none), and to position robots relative to the ball so they don't have to travel as far, for example if we are stopping while setting up for a corner kick in the enemy half.

### Testing Done

local testing

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by garethellis0** [comment]
Looks like `STPTest_test_only_test_plays_are_registered_in_play_factory_Test` is failing in CI

File: src/thunderbots/software/ai/hl/stp/play/halt_play.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+bool HaltPlay::isApplicable(const World &world) const
+{
+    return world.gameState().isHalted();
+}
+
+bool HaltPlay::invariantHolds(const World &world) const
+{
+    return world.gameState().isHalted();
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/play/halt_play.cpp:20]: Can you also add gamestate checks to the other existing plays to make sure that they are _not_ applicable if the game state is not correct?

File: src/thunderbots/software/ai/hl/stp/play/halt_play.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+void HaltPlay::getNextTactics(TacticCoroutine::push_type &yield)
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/play/halt_play.cpp:23]: As a note for the future, we should make this take the number of robots to stop in the constructor.

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada: - main thing is updating the other plays to make sure they can't run if we're in a halt state (might be easier to whitelist conditions instead of blacklisting, I'm not sure), but I'll leave it up to you whether or not we do that there.

File: src/thunderbots/software/ai/hl/stp/play/halt_play.cpp (hunk 2 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/play/halt_play.cpp:23]: why? The plays don't have much control over their construction, it's done automatically during play selection

File: src/thunderbots/software/ai/hl/stp/play/halt_play.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/play/halt_play.cpp:20]: Will do in a separate PR

**Review by garethellis0 — APPROVED** [review]

**Pull request merged**